### ISSUE TITLE
URLs containing highlighted search terms incorrectly include markup in href attribute

### ISSUE THREAD

**Issue by noidexe** [RocketChat/Rocket.Chat#28368]
### Description:

If you click on a URL that has any part of it highlighted (due to being result of a search), the actual href attribute will include <mark>search_term</mark> which causes an error when trying to follow the link

### Steps to reproduce:

1. Go to search
2. Type ".com" or anything that you know will be part of a URL in the search results.
3. Click on any URL including the highlighted search term

### Expected behavior:

You are able to follow the link properly

### Actual behavior:

The link incorrectly includes <mark></mark> tags arround the highlighted search term
![image](https://user-images.githubusercontent.com/526829/224405665-68c8d332-638f-4269-8d3b-a87e7ecb3b61.png)

### Server Setup Information:

- Version of Rocket.Chat Server: 5.1
If you need extra info I'll have to request it to the server admin

### Client Setup Information

- Browser Version
- Manjaro Linux

**Comment by garvitkhandelwal** [comment]
@noidexe Are you not able to navigate to the URL(with highlighted text) properly? Asking because I'm able to navigate.

**Comment by noidexe** [comment]
You're right. I forgot to search among the closed issues https://github.com/RocketChat/Rocket.Chat/issues/23285
